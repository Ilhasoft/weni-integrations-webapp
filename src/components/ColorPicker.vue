<template>
  <div class="color-picker">
    <div
      class="color-picker__color"
      v-for="color in colors"
      :key="color"
      :style="{ 'background-color': color }"
      @click="emitColor(color)"
    />
    <label class="color-picker__input">
      <unnnic-icon-svg class="color-picker__input__icon" icon="add-circle-1" size="sm" />
      <input ref="color-picker__input" type="color" @change="addColor()" />
    </label>
  </div>
</template>

<script>
  export default {
    name: 'color-picker',
    props: {
      initialColors: {
        type: Array,
        default: () => ['#009E96', '#262626', '#C4EAF5', '#C4C4C4'],
      },
    },
    data() {
      return {
        colors: this.initialColors,
      };
    },
    methods: {
      addColor(event) {
        const color = event.target.value;
        this.colors.push(color);
        this.emitColor(color);
      },
      emitColor(color) {
        this.$emit('colorChange', color);
      },
    },
  };
</script>

<style lang="scss" scoped>
  .color-picker {
    display: flex;
    gap: $unnnic-inline-xs;

    &__color {
      width: $unnnic-icon-size-xs;
      height: $unnnic-icon-size-xs;
      border-radius: 50%;
    }

    &__input {
      width: $unnnic-icon-size-xs;
      height: $unnnic-icon-size-xs;
      border-radius: 50%;

      &__icon {
        left: -2px;
        top: -6px;
      }

      input {
        visibility: hidden;
      }
    }
  }
</style>
