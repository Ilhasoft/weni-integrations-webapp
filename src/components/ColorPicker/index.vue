<template>
  <div class="color-picker">
    <div
      class="color-picker__color"
      v-for="color in colors"
      :key="color"
      :style="{ 'background-color': color }"
      @click="emitColor(color)"
    />
    <label class="color-picker__input">
      <unnnic-icon-svg class="color-picker__input__icon" icon="add-circle-1" size="sm" ref="icon" />
      <input ref="color-picker__input" type="color" @change="addColor" />
    </label>
  </div>
</template>

<script>
  export default {
    name: 'color-picker',
    props: {
      initialColors: {
        type: Array,
        default: () => ['#009E96', '#262626', '#C4EAF5', '#C4C4C4'],
      },
    },
    data() {
      return {
        colors: this.initialColors,
      };
    },
    methods: {
      addColor(event) {
        const color = event.target.value;
        this.colors.push(color);
        this.emitColor(color);
      },
      emitColor(color) {
        this.$emit('colorChange', color);
      },
    },
  };
</script>

<style lang="scss" scoped>
  @import './styles.scss';
</style>
