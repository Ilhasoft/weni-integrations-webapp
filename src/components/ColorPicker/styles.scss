.color-picker {
    display: flex;
    gap: $unnnic-inline-xs;

    &__color {
      width: $unnnic-icon-size-xs;
      height: $unnnic-icon-size-xs;
      border-radius: $unnnic-border-radius-pill;
    }

    &__input {
      width: $unnnic-icon-size-xs;
      height: $unnnic-icon-size-xs;
      border-radius: $unnnic-border-radius-pill;

      &__icon {
        left: -2px;
        top: -6px;
      }

      input {
        width: 0;
        height: 0;
        visibility: hidden;
      }
    }
  }