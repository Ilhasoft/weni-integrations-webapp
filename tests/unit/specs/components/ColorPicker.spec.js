import { mount, createLocalVue } from '@vue/test-utils';
import ColorPicker from '@/components/ColorPicker/index.vue';

const localVue = createLocalVue();

describe('ColorPicker.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(ColorPicker, {
      localVue,
      stubs: {
        UnnnicIconSvg: true,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should add a new color', () => {
    const event = {
      target: { value: '#ff00ff' },
    };
    expect(wrapper.vm.colors).toHaveLength(4);

    wrapper.vm.addColor(event);
    expect(wrapper.vm.colors).toHaveLength(5);
    expect(wrapper.vm.colors.pop()).toEqual(event.target.value);
  });

  it('should emit new color', () => {
    const color = '#ff00ff';
    wrapper.vm.emitColor(color);
    expect(wrapper.emitted('colorChange')).toBeTruthy();
    expect(wrapper.emitted('colorChange')[0]).toEqual([color]);
  });
});
