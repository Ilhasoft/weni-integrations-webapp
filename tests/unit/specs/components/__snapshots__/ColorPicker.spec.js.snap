// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker.vue should be rendered properly 1`] = `
<div class="color-picker">
  <div class="color-picker__color" style="background-color: rgb(0, 158, 150);"></div>
  <div class="color-picker__color" style="background-color: rgb(38, 38, 38);"></div>
  <div class="color-picker__color" style="background-color: rgb(196, 234, 245);"></div>
  <div class="color-picker__color" style="background-color: rgb(196, 196, 196);"></div> <label class="color-picker__input">
    <unnniciconsvg-stub icon="add-circle-1" size="sm" class="color-picker__input__icon"></unnniciconsvg-stub> <input type="color">
  </label>
</div>
`;
